### PR TITLE
fix: Cleaning up #5232

### LINKED ISSUE #5089: Imported exclude block not working on single unit commands
## Describe the bug

When using include to import exclude blocks from a parent configuration, the exclusion rules are ignored for single unit runs but work correctly for `--all` runs.

## Steps To Reproduce

The issue can be reproduced with this minimal setup:

**Directory structure**

```sh
.
├── module
│   └── main.tf
├── root.hcl
└── unit
    └── terragrunt.hcl
```

**File contents**

`root.hcl`
```hcl
# Example to exclude all units from running
exclude {
  if                   = true    # Boolean to determine exclusion.
  no_run               = true    # Boolean to prevent the unit from running (even when not using `--all`).
  actions              = ["all"] # List of actions to exclude (e.g., "plan", "apply", "all", "all_except_output").
  exclude_dependencies = true    # Boolean to determine if dependencies should also be excluded.
}
```

`unit/terragrunt.hcl`
```hcl
# Include all settings from root.hcl
include "root" {
  path = find_in_parent_folders("root.hcl")
}

# Use example module
terraform {
  source = "../module"
}

inputs = {}
```

`module/main.tf`
```hcl
output "example_output" {
  value = "hello world"
}
```

**Reproduction commands**

```sh
# Switch to unit directory
$ cd unit/

# This should be excluded (not working as expected)
$ terragrunt run -- plan
Initializing the backend...
Initializing provider plugins...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

Changes to Outputs:
  + example_output = "hello world"

You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.

─────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.

# works as expected
$ terragrunt run --all -- plan
❯❯ Run Summary  1 units  292ns
   ────────────────────────────
   Excluded     1

```

When I change the content of `unit/terragrunt.hcl` to include the exclude block directly:

```hcl
# Include all settings from root.hcl
include "root" {
  path = find_in_parent_folders("root.hcl")
}

# Example to exclude all units from running
exclude {
  if                   = true    # Boolean to determine exclusion.
  no_run               = true    # Boolean to prevent the unit from running (even when not using `--all`).
  actions              = ["all"] # List of actions to exclude (e.g., "plan", "apply", "all", "all_except_output").
  exclude_dependencies = true    # Boolean to determine if dependencies should also be excluded.
}

# Use example module
terraform {
  source = "../module"
}

inputs = {}
```

Running the commands again produces the expected behavior, but is not DRY:

```sh
# Produces no output
$ terragrunt run -- plan

$ terragrunt run --all -- plan

❯❯ Run Summary  1 units  291ns
   ────────────────────────────
   Excluded     1
```

## Expected behavior

Included exclude block configurations should be respected regardless of which command is used.

## Versions

- Terragrunt version: v0.93.6
- OpenTofu/Terraform version: Terraform v1.13.5 on darwin_arm64
- Environment details: macOS 15.7.2 - zsh 5.9 (arm-apple-darwin24.0.0)

### PR DESCRIPTION
**Pull request by yhakbar**
## Description

Fixes #5089.

Closes #5232.

The implementation in #5232 was correct, but it didn't include tests for the behavior being fixed, and it also didn't include coverage of other blocks that might also be included from separate config.

This pulls in the fix from #5232 and adds tests for the behavior being fixed.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] I authored this code entirely myself
- [x] I am submitting code based on open source software (e.g. MIT, MPL-2.0, Apache)
- [x] I am adding or upgrading a dependency or adapted code and confirm it has a compatible open source license
- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed configuration inheritance for exclude blocks. When a unit includes a parent configuration without defining its own exclude block, the parent's exclude settings are now properly merged into the unit configuration instead of being discarded.

* **Tests**
  * Added comprehensive test coverage for configuration inheritance and merging behavior across multiple configuration blocks.

### CODE REVIEW

File: pkg/config/exclude_include_test.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+			assert.Equal(t, []string{"plan", "apply"}, parsed.Exclude.Actions)
+			require.NotNil(t, parsed.Exclude.NoRun)
+			assert.True(t, *parsed.Exclude.NoRun)
+		})
+	}
+}
+
+// Child-defined exclude blocks must still take precedence over the included
+// parent's exclude block, regardless of merge strategy.
+func TestParseConfig_ChildExcludeOverridesIncludedConfig(t *testing.T) {
```
Comment (denis256) [inline pkg/config/exclude_include_test.go:83]: Makes sense to add a test for  child with `exclude { if = false }`

File: pkg/config/exclude_include_test.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
+package config_test
```
Comment (denis256) [inline pkg/config/exclude_include_test.go:1]: Missing test for empty `exclude {}` block on child, it may trigger different nil/zero errors

**Review by denis256 — APPROVED** [review]
